### PR TITLE
FIX: bindings presentation for control schemes (ISX-1132)

### PR DESCRIPTION
**Pull request by ritamerkl**
### Description

Updated the action tree view to only show bindings that are included into the current selected control scheme. Also I corrected the {GLOBAL} identifier to only show up if no control scheme is assigned to a binding. The ticket can be found [here](https://jira.unity3d.com/browse/ISX-1132), see the comments for details.

### Changes made

fixed:
- hiding bindings assigned to other control schemes if a control scheme is selected
- showing global identifier only on bindings that are not assigned to any control scheme

### Notes

As the feature seems not to work reliable good in the old UI there were misunderstandings on how this feature has to work, so a different solution was implemented before.

### Checklist

Before review:

- [ ] Changelog entry added.
    - Explains the change in `Changed`, `Fixed`, `Added` sections.
    - For API change contains an example snippet and/or migration example.
    - FogBugz ticket attached, example `([case %number%](https://issuetracker.unity3d.com/issues/...))`.
    - FogBugz is marked as "Resolved" with *next* release version correctly set.
- [ ] Tests added/changed, if applicable.
    - Functional tests `Area_CanDoX`, `Area_CanDoX_EvenIfYIsTheCase`, `Area_WhenIDoX_AndYHappens_ThisIsTheResult`.
    - Performance tests.
    - Integration tests.
- [ ] Docs for new/changed API's.
    - Xmldoc cross references are set correctly.
    - Added explanation how the API works.
    - Usage code examples added.
    - The manual is updated, if needed.

During merge:

- [x] Commit message for squash-merge is prefixed with one of the list:
    - `NEW: ___`.
    - `FIX: ___`.
    - `DOCS: ___`.
    - `CHANGE: ___`.
    - `RELEASE: 1.1.0-preview.3`.

### CODE REVIEW

File: Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/Views/ActionsTreeView.cs [hunk 1]
```diff
@@ -351,28 +358,41 @@ public static List<TreeViewItemData<ActionOrBindingData>> GetActionsAsTreeViewDa
...
             return name;
         }
 
-        private static bool IsBindingPartOfCurrentControlScheme(SerializedInputBinding serializedInputBinding, InputControlScheme? currentControlScheme, int currentControlSchemeIndex)
+        private static bool IsBindingAssignedToNoOrAllControlSchemes(SerializedInputBinding serializedInputBinding, SerializedProperty allControlSchemes, InputControlScheme? currentControlScheme)
         {
-            if (currentControlScheme.HasValue && currentControlSchemeIndex >= 0)
+            if (allControlSchemes.arraySize <= 0 || !currentControlScheme.HasValue || string.IsNullOrEmpty(currentControlScheme.Value.name))
+                return false;
+            if (serializedInputBinding.controlSchemes.Length <= 0)
```
Comment (jamesmcgill) [inline Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/Views/ActionsTreeView.cs:375]: This looks like the 'AssignedToNo ControlSchemes' part.
But where does it get that it's assigned to 'All' ControlSchemes?

File: Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/Views/ActionsTreeView.cs [hunk 2]
```diff
@@ -351,28 +358,41 @@ public static List<TreeViewItemData<ActionOrBindingData>> GetActionsAsTreeViewDa
...
+        private static bool IsBindingPartOfCurrentControlScheme(SerializedInputBinding serializedInputBinding, InputControlScheme? currentControlScheme)
+        {
+            if (currentControlScheme.HasValue && !string.IsNullOrEmpty(currentControlScheme.Value.name))
+            {
+                if (serializedInputBinding.controlSchemes.Length <= 0)
+                    return true;
                 return serializedInputBinding.controlSchemes.Contains(currentControlScheme.Value.name);
+            }
 
             return true;
```
Comment (jamesmcgill) [inline Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/Views/ActionsTreeView.cs:390]: Should this be `return false`?

File: Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/Views/ActionsTreeView.cs [hunk 3]
```diff
@@ -351,28 +358,41 @@ public static List<TreeViewItemData<ActionOrBindingData>> GetActionsAsTreeViewDa
...
+            if (serializedInputBinding.controlSchemes.Length <= 0)
+                return true;
+            return false;
+        }
+
+        private static bool IsBindingPartOfCurrentControlScheme(SerializedInputBinding serializedInputBinding, InputControlScheme? currentControlScheme)
+        {
+            if (currentControlScheme.HasValue && !string.IsNullOrEmpty(currentControlScheme.Value.name))
+            {
+                if (serializedInputBinding.controlSchemes.Length <= 0)
```
Comment (jamesmcgill) [inline Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/Views/ActionsTreeView.cs:385]: Maybe I don't understand this line (as it's identical to the one I commented on above).

Doesn't this mean the binding has no control schemes? Should it return `false` in that case?
Maybe needs a comment or a function to help clear up what it does, if it's meaning is more complex.

**Review by jamesmcgill — APPROVED** [review]
Looks good, added a couple of comments.
Hold off merging until after the release please

File: Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/Views/ActionsTreeView.cs (hunk 1 above)
Comment (ritamerkl) [reply Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/Views/ActionsTreeView.cs:375]: There was confusion about the {GLOBAL} identifier showing up. Actually it turned out that the identifier should only show up if the control scheme is not assigned to any control scheme. I forgot to rename the method, I updated the name now. @jamesmcgill

File: Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/Views/ActionsTreeView.cs (hunk 2 above)
Comment (ritamerkl) [reply Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/Views/ActionsTreeView.cs:390]: I updated the method name to be more descriptive & added comments @jamesmcgill

File: Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/Views/ActionsTreeView.cs (hunk 3 above)
Comment (ritamerkl) [reply Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/Views/ActionsTreeView.cs:385]: I updated the method name to be more descriptive & added comments

**Pull request merged**